pkg/virt-handler/vm.go

```diff
@@ -2839,6 +2839,8 @@ func (d *VirtualMachineController) vmUpdateHelperMigrationTarget(origVMI *v1.Vir
 		}
 		log.Log.Object(vmi).Infof("Signaled target pod for failed migration to clean up")
 		// nothing left to do here if the migration failed.
+		// Re-enqueue to trigger handler final cleanup
+		d.queue.AddAfter(controller.VirtualMachineInstanceKey(vmi), time.Second)
```

Comment:
Thank you @fossedihelm!

I wonder: is there a reason to wait for the target pod to [panic](https://github.com/kubevirt/kubevirt/blob/55bde60ce16453c53e2ad13bc532322739a11642/cmd/virt-launcher/virt-launcher.go#L269-L272) before triggering the handler's cleanup? In other words, WDYT about calling `processVmCleanup()` here instead? My reasoning is that we're going to kill this pod in a brutal way anyways, so I'm not sure if there's a reason to wait for it to actually shut down.

WDYT?

Reply:
FWIW I'm not against your change, just trying to think if there's a better alternative.
In terms of scale & performance, this change might cause some overhead on a mass migration cancellation, but I would assume the impact is small.

Reply:
Sure! Thank you for the review and the discussion

Reply:
It is for sure something we can think about

Reply:
> FWIW I'm not against your change, just trying to think if there's a better alternative. In terms of scale & performance, this change might cause some overhead on a mass migration cancellation, but I would assume the impact is small.

We have bounds per node and cluster for how many migrations are happening. IMHO the scale is not a problem here as we are talking in single digits...

>@xpivarc any insights?

The reconciliation and trigger of `processVmCleanup` should still go through all the checks necessary... That is only going to be enforce a) by the requeue (that is only speeding up the checks) b) making sure `processVmCleanup` is self-contained and contains all the checks which it does not today.

Reply:
@iholder101 Maybe the `processVmCleanup` needs to wait until kubelet will finish the pod termination first

Reply:
It should wait for the launcher to exit and kill it if it is not exiting in time... otherwise the cleanup will fail